.github/workflows/client.yml

```diff
@@ -0,0 +1,42 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows 
+on:  
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true # ignore the failure of a step and avoid terminating the job
+      - name: Run Docker build
+        run: |
+          docker build \
+            --target gobuild \
+            --tag go-build-env .
+          docker build \
+            --tag keep-client . 
+      - name: Create test results directory
+        run: |
+          mkdir test-results
+      - name: Run Go tests
+        run: |
+          docker run \
+            --volume $GITHUB_WORKSPACE/test-results:/mnt/test-results \
+            --workdir /go/src/github.com/keep-network/keep-core \
+            go-build-env \
+            gotestsum --junitfile /mnt/test-results/unit-tests.xml
+      - name: Publish unit test results
+        uses: EnricoMi/publish-unit-test-result-action@v1.7
+        if: always() # guarantees that this action always runs, even if earlier steps fail
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          files:  ./test-results/unit-tests.xml
+          check_name: Go Test Results # name under which test results will be presented in GitHub (optional)
+          comment_on_pr: false # turns off commenting on Pull Requests
```

Comment:
If things are working correctly, I believe the check should already be bubbling the important parts of this information up, including line annotations, e.g. like https://github.com/keep-network/keep-ecdsa/actions/runs/555260607 . I don't think we need the junit intermediary here, see https://github.com/actions/toolkit/blob/master/docs/commands.md#problem-matchers and the setup-go action's problem matcher at https://github.com/actions/setup-go/blob/main/matchers.json .

Reply:
(Of note: we may want to add our own matcher that works with our go-to test output.)